src/main/java/com/minecolonies/coremod/entity/ai/citizen/shepherd/EntityAIWorkShepherd.java

```diff
@@ -0,0 +1,504 @@
+    }
+
+    /**
+     * Shears a sheep, with a chance of dying it!
+     *
+     * @return the wanted AiState
+     */
+    private AIState shearSheep()
+    {
+        job.removeDeadSheep();
+
+        final List<EntitySheep> sheepies = job.getSheep();
+
+        if (sheepies.isEmpty())
+        {
+            return SHEPHERD_SEARCH_FOR_SHEEP;
+        }
+
+        final EntitySheep sheep = sheepies.stream().filter(sheepie -> !sheepie.getSheared()).findFirst().orElse(null);
+
+        if (!equipShears())
+        {
+            return PREPARING;
+        }
+
+        if (worker.getHeldItemMainhand() != null && sheep != null)
+        {
+            final List<ItemStack> items = sheep.onSheared(worker.getHeldItemMainhand(),
+              worker.worldObj,
+              worker.getPosition(),
+              net.minecraft.enchantment.EnchantmentHelper.getEnchantmentLevel(net.minecraft.init.Enchantments.FORTUNE, worker.getHeldItemMainhand()));
+
+            dyeSheepChance(sheep);
+
+            worker.getHeldItemMainhand().damageItem(1, worker);
+
+            worker.addExperience(EXP_PER_SHEEP);
+
+            for (final ItemStack item : items)
+            {
+                worker.getInventoryCitizen().addItemStackToInventory(item);
+            }
+        }
+
+        //setDelay(40);
+
+        return SHEPHERD_DECIDE;
+    }
+
+    /**
+     * Possibly dyes a sheep based on their Worker Hut Level
+     */
+    private void dyeSheepChance(final EntitySheep sheep)
+    {
+        if (worker.getWorkBuilding() != null)
+        {
+            final int chanceToDye = worker.getWorkBuilding().getBuildingLevel();
+
+            final int rand = world.rand.nextInt(100);
+
+            if (rand <= chanceToDye)
+            {
+                final int dyeInt = world.rand.nextInt(15);
+                sheep.setFleeceColor(EnumDyeColor.byMetadata(dyeInt));
+            }
+        }
+    }
+
+    private AIState breedSheep()
+    {
+        job.removeDeadSheep();
+
+        final List<EntitySheep> sheepies = job.getSheep();
+
+        final EntitySheep sheepOne = sheepies.stream().filter(sheep -> sheep.getGrowingAge() == 0).findAny().orElse(null);
+
+        if (sheepOne == null)
+        {
+            return PREPARING;
+        }
+
+        final EntitySheep sheepTwo = sheepies.stream().filter(sheep ->
+          {
+              final float range = sheep.getDistanceToEntity(sheepOne);
+              final boolean isSheepOne = sheepOne.equals(sheep);
+              return sheep.getGrowingAge() == 0 && range <= DISTANCE_TO_BREED && !isSheepOne;
+          }
+        ).findAny().orElse(null);
+
+        if (sheepTwo == null)
+        {
+            return PREPARING;
+        }
+
+        if (!equipWheat())
+        {
+            return PREPARING;
+        }
+
+        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.shepherd.breedingSheep"));
+
+        breedTwoSheep(sheepOne, sheepTwo);
+
+        //setDelay(40);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)